modules/shapefile/src/lib/parsers/parse-shp-geometry.js

```diff
@@ -215,3 +231,35 @@ function concatPositions(xyPositions, mPositions, zPositions) {
 
   return positions;
 }
+
+/**
+ * Returns the direction of the polygon path
+ * A positive number is clockwise.
+ * A negative number is counter clockwise.
+ *
+ * @param  {Float32Array} positions
+ * @return {Number}           [description]
+ */
+function getWindingDirection(positions) {
+  return Math.sign(getSignedArea(positions));
+}
+
+/**
+ * Get signed area of flat typed array of 2d positions
+ *
+ * @param  {Float64Array} positions [description]
+ * @return {Number}           [description]
+ */
+function getSignedArea(positions) {
+  let area = 0;
+
+  // Rings are closed according to shapefile spec
+  const nCoords = positions.length / 2 - 1;
+  for (let i = 0; i < nCoords; i++) {
+    const p1 = positions.subarray(i * 2, (i + 1) * 2);
```

Comment:
You are effectively constructing a lot of small objects Here which is exactly what you wanted to avoid.

Reply:
I thought views into the typed array would be cheap, but I guess the bottleneck is creating the JS object itself, not a data copy?

Reply:
Anyways, switched to not create subarrays.